backend/ecs_tasks/delete_files/parquet_handler.py

```diff
@@ -51,15 +81,22 @@ def delete_from_table(table, to_delete):
     for column in to_delete:
```

Comment:
nit: the little bit of repetition here suggests to me that `to_delete` should contain both normal and composite deletions, then we should determine the correct function call based on the deletion type e.g.

```
for column in to_delete:
        is_composite = # some way of finding out if it's a composite or simple
        get_row_indexes = get_row_indexes_to_delete if is_composite else get_row_indexes_to_delete_for_composite
        indexes = get_row_indexes(
            table, column["Columns"], column["MatchIds"]
        )
        table = table.filter(~indexes)
```

Reply:
not a huge deal if it would be problematic to do it like this, but if it's possible we should consider it

Reply:
You're right. I went on extending the `Columns` object to contain a `Type` property for each match, and that allowed me to simplify the code a lot in most of the lambdas and the ecs task too! Thanks